packages/hint-manifest-scoped-urls/src/hint.ts

```diff
@@ -0,0 +1,131 @@
+            let networkData: NetworkData;
+
+            try {
+                networkData = await context.fetchContent(startUrl);
+            } catch (e) {
+                debug(`Failed to fetch ${startUrl}`);
+                const message = `Request failed for 'start_url'`;
+
+                context.report(resource, message, { location: startUrllocation });
+
+                return false;
+            }
+
+            const response = networkData.response;
+
+            if (response.statusCode !== 200) {
+                const message = `Specified 'start_url' is not accessible. (status code: ${response.statusCode}).`;
+
+                context.report(resource, message, { location: startUrllocation });
+
+                return false;
+            }
+
+            return true;
+        };
+
+        /**
+         *
+         * @param property `property` to be found in the Manifest
+         * @param parsedContent Manifest
+         * @param resource
+         */
+        const manifestPropertyFound = (property: string, parsedContent: Manifest, resource: string): boolean => {
+            const found = parsedContent.hasOwnProperty(property);
+
+            if (!found) {
+                const message = `Property '${property}' not found in manifest file`;
+
+                context.report(resource, message);
+
+                return false;
+            }
+
+            return true;
+        };
+
+        /**
+         * Checks that the `start_url` is under the scope of
+         * the URL specified in the `scope`
+         * @param parsedContent
+         * @param resource
+         */
+        const startUrlInScope = (parsedContent: Manifest, resource: string, startUrllocation: ProblemLocation): boolean => {
+            const scope = parsedContent.scope || '/';
+            const startURL = parsedContent.start_url;
+            const relativePath = relative(scope, startURL!);
+            const inScope = relativePath && !relativePath.startsWith('..');
+
+            if (relativePath === '') {
+                return true;
+            }
+
+            if (!inScope) {
+                const message = `'start_url' is not in scope of the app.`;
+
+                context.report(resource, message, { location: startUrllocation });
+
+                return false;
+            }
+
+            return true;
+        };
+
+        const validate = async ({ getLocation, parsedContent, resource }: ManifestParsed) => {
+            const resourceURL = new URL(resource);
+            const hostnameWithProtocol = `${resourceURL.protocol}//${resourceURL.host}`;
+
+            debug(`Validating hint manifest-scoped-urls`);
+
+            const hasStartUrl = manifestPropertyFound('start_url', parsedContent, resource);
+
+            if (hasStartUrl) {
+                const startUrlLocation = getLocation('start_url');
+                const notSameOrigin = parsedContent.start_url!.startsWith('http');
```

Comment:
Technically more of an `isRelative` check, which seems appropriate [according to the MDN documentation](https://developer.mozilla.org/en-US/docs/Web/Manifest#start_url), but probably should cover non-http protocols too.

@molant Do you recall if we have a general utility already to check if an URL is absolute/relative?

Reply:
@antross  Just realized this would fail for something like `httpLover.html`.

Changing it to 
```js
const notSameOrigin = parsedContent.start_url.startsWith('http://') || parsedContent.start_url.startsWith('https://');
```

Reply:
I think you can use `is-regular-protocol` for this. It comes in `@hint/utils`.

```ts
import { isRegularProtocol } from '@hint/utils/dist/src/network/is-regular-protocol';

const notSameOrigin = isRegularProtocol(parsedContent.start_url)
```